experimental/delegate/mlp_aie_bf16_plugin.cpp

```diff
@@ -109,8 +109,8 @@ using B_DATATYPE = bfloat16_t;
 using C_DATATYPE = float; // bfloat16_t;
 
 // Types of the matmul LHS, RHS, and result, as seen by the model
-using ModelLhsDType = bfloat16_t;
-using ModelRhsDType = bfloat16_t;
+using ModelLhsDType = float; // bfloat16_t;
+using ModelRhsDType = float; // bfloat16_t;
```

Comment:
hmmm, I was expecting you to convert the input buffers to bf16 manually some where for this kernel?

Reply:
ah I see the kernel types are defined right above.

Reply:
The magic of templates!  :-D  The converter/copier classes do that automatically, and those decisions are made at compile time during template param binding.